packages/gatsby/cache-dir/ensure-resources.js

```diff
@@ -74,6 +74,16 @@ class EnsureResources extends React.Component {
   }
 
   render() {
+    if (!this.state.pageResources) {
+      console.warn(
+        `EnsureResources was not able to find resources for path: "${this.props.location.pathname}"`,
+        `This typically means that an issue occurred building components for that path`
+      )
+      throw new Error(
+        `EnsureResources was not able to find resources for path: "${this.props.location.pathname}"`
```

Comment:
Let's not duplicate this line in warning and error message - I would vote to get rid of warn completely and just put this whole text in thrown error - sounds reasonable?

Reply:
Ah I was mostly thinking about production builds here - maybe we should have conditional logic for `gatsby develop` (using `process.env.NODE_ENV === `production` to show short(ish) message, but for `gatsby develop` case we can show cleaning up `.cache` (but it probably should be both `.cache` and `public` given stale page-data being on of potential reasons for this)